pkg/spoke/controllers/appliedmanifestcontroller/appliedmanifestwork_controller.go

```diff
@@ -201,19 +201,21 @@ func (m *AppliedManifestWorkController) syncManifestWork(
 }
 
 // findUntrackedResources returns applied resources which are no longer tracked by manifestwork
+// API version should be ignored when checking if a resource is no longer tracked by a manifestwork.
+// This is because we treat resources of same GroupResource but different version equivalent.
 func findUntrackedResources(appliedResources, newAppliedResources []workapiv1.AppliedManifestResourceMeta) []workapiv1.AppliedManifestResourceMeta {
 	var untracked []workapiv1.AppliedManifestResourceMeta
 
 	resourceIndex := map[workapiv1.AppliedManifestResourceMeta]struct{}{}
 	for _, resource := range newAppliedResources {
 		key := resource.DeepCopy()
-		key.UID = ""
```

Comment:
why was this removed?

Reply:
I do not think we need to empty the uid. For the newAppliedResources, we also get its uid by Get before calling findUntrackedResources.  And I think uid will not change when resource version is changed. 

If the uid in newAppliedResource is different than what is recorded in appliemanifestwork, we will regard the recorded one in appliemanifestwork as untracked, and delete the resource with the uid of the recorded one as a precondition (https://github.com/open-cluster-management/work/blob/11f530f467683e5b8d6daaff80d32fad97cf54b8/pkg/helper/helpers.go#L216). It should also be fine because we will get an IsNotFound error when delete it.

I will update the doc to explain it.

Reply:
> and delete the resource with the uid of the recorded one as a precondition

does this result in controller fighting during shared resource situations where each one sees a mismatching UID?

Reply:
I think even in shared resource situation, the uid recorded in two appliemanifestworks will be the same, because, we get the current uid of the resource and record it into appliemanifestworks before deleting it.

Mismatch happened when user creates manifestwork/foo at first, delete it and then creates   manifestwork/bar with a shared resource immediately.  appliedmanifestwork/bar will recreate the resource and appliedmanifestwork/foo will record a mismatched UID. In this case, the resource will be deleted with a mismatched uid, return a NotFound error and the error will be ignored.

Reply:
> The shared resource will be deleted.

This is only the case until you create the orphaning policy.



> then creates manifestwork/bar with a shared resource immediately. appliedmanifestwork/bar will recreate the resource and appliedmanifestwork/foo will record a mismatched UID. In this case, the resource will be deleted with a mismatched uid, return a NotFound error and the error will be ignored. The newly created resource by manifestwork/bar will not be deleted.

Is adoption still possible?  The case where you create an instance and then later create a matching manifest work.